types/instance.d.ts

```diff
@@ -353,4 +353,25 @@ export interface FastifyInstance<
    * Prints the representation of the internal radix tree used by the router
    */
   printRoutes(): string;
+
+  /**
+   *  Frozen read-only object registering the initial options passed down by the user to the fastify instance
+   */
+  initialConfig: Readonly<{
+    connectionTimeout?: number,
+    keepAliveTimeout?: number,
+    bodyLimit?: number,
+    caseSensitive?: boolean,
+    http2?: boolean,
+    https?: boolean | { allowHTTP1: boolean },
+    ignoreTrailingSlash?: boolean,
+    disableRequestLogging: boolean,
+    maxParamLength?: number,
+    onProtoPoisoning?: 'error' | 'remove' | 'ignore',
+    onConstructorPoisoning?: 'error' | 'remove' | 'ignore',
+    pluginTimeout?: number,
+    requestIdHeader?: string,
+    requestIdLogLabel: string,
+    http2SessionTimeout: number
```

Comment:
shouldn't all the properties be optional?

Reply:
Yes - will update